packages/clerk-js/src/core/resources/Session.ts

```diff
@@ -351,28 +353,60 @@ export class Session extends BaseResource implements SessionResource {
     }
 
     const tokenId = this.#getCacheId(template, organizationId);
+
     const cachedEntry = skipCache ? undefined : SessionTokenCache.get({ tokenId }, leewayInSeconds);
 
     // Dispatch tokenUpdate only for __session tokens with the session's active organization ID, and not JWT templates
     const shouldDispatchTokenUpdate = !template && organizationId === this.lastActiveOrganizationId;
 
     if (cachedEntry) {
+      debugLogger.debug(
+        'Using cached token (no fetch needed)',
+        {
+          tokenId,
+        },
+        'session',
+      );
       const cachedToken = await cachedEntry.tokenResolver;
       if (shouldDispatchTokenUpdate) {
         eventBus.emit(events.TokenUpdate, { token: cachedToken });
       }
```

Comment:
🤔 I'm wondering why we wanted to store the promise in the first place, instead of just the token. I suppose it gets something into the cache faster.